examples/escrow/programs/escrow/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+            ctx.accounts.into_set_authority_context(signer),
+            AuthorityType::AccountOwner,
+            Some(ctx.accounts.escrow_account.initializer_key),
+        )?;
+
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(initializer_amount: u64)]
+pub struct InitializeEscrow<'info> {
+    #[account(signer)]
+    pub initializer: AccountInfo<'info>,
+    #[account(mut,
+        constraint = initializer_deposit_token_account.amount >= initializer_amount
+    )]
+    pub initializer_deposit_token_account: CpiAccount<'info, TokenAccount>,
+    pub initializer_receive_token_account: CpiAccount<'info, TokenAccount>,
+    #[account(init)]
+    pub escrow_account: ProgramAccount<'info, EscrowAccount>,
+    pub token_program: AccountInfo<'info>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+#[derive(Accounts)]
+pub struct Exchange<'info> {
+    #[account(signer)]
+    pub taker: AccountInfo<'info>,
+    #[account(mut)]
+    pub taker_deposit_token_account: CpiAccount<'info, TokenAccount>,
+    #[account(mut)]
+    pub taker_receive_token_account: CpiAccount<'info, TokenAccount>,
+    #[account(mut)]
+    pub pda_deposit_token_account: CpiAccount<'info, TokenAccount>,
+    #[account(mut)]
+    pub initializer_receive_token_account: CpiAccount<'info, TokenAccount>,
+    #[account(mut)]
+    pub initializer_main_account: AccountInfo<'info>,
+    #[account(mut,
+        constraint = escrow_account.taker_amount <= taker_deposit_token_account.amount,
+        constraint = escrow_account.initializer_deposit_token_account == *pda_deposit_token_account.to_account_info().key,
+        constraint = escrow_account.initializer_receive_token_account == *initializer_receive_token_account.to_account_info().key,
+        constraint = escrow_account.initializer_key == *initializer_main_account.key,
+        close = initializer_main_account
+    )]
+    pub escrow_account: ProgramAccount<'info, EscrowAccount>,
+    pub pda_account: AccountInfo<'info>,
+    pub token_program: AccountInfo<'info>,
+}
+
+#[derive(Accounts)]
+pub struct CancelEscrow<'info> {
+    pub initializer: AccountInfo<'info>,
+    #[account(mut)]
+    pub pda_deposit_token_account: CpiAccount<'info, TokenAccount>,
+    pub pda_account: AccountInfo<'info>,
+    #[account(mut,
+        constraint = escrow_account.initializer_key == *initializer.key,
+        constraint = escrow_account.initializer_deposit_token_account == *pda_deposit_token_account.to_account_info().key,
+        close = initializer
+    )]
+    pub escrow_account: ProgramAccount<'info, EscrowAccount>,
+    pub token_program: AccountInfo<'info>,
+}
+
+#[account]
+pub struct EscrowAccount {
+    pub initializer_key: Pubkey,
+    pub initializer_deposit_token_account: Pubkey,
+    pub initializer_receive_token_account: Pubkey,
+    pub initializer_amount: u64,
+    pub taker_amount: u64,
+}
+
+impl<'a, 'b, 'c, 'info> From<&mut InitializeEscrow<'info>>
+    for CpiContext<'a, 'b, 'c, 'info, SetAuthority<'info>>
```

Comment:
Tip. You can use the anonymous lifetime to make this more concise like this `CpiContext<'_, '_, '_, 'info, SetAuthority<'info>>` (allowing one to remove the declared lifetimes above).

Reply:
Just wanted to confirm this comment was for the functions below (which made it cleaner) and not this trait implementation, since didn't seem like it could work with CpiContext.

Reply:
It should work for both. DId you remove the lifetimes from the impl alltogether? Example here https://github.com/project-serum/anchor/blob/armani/cfo/examples/cfo/programs/cfo/src/lib.rs#L587.

Reply:
Thanks for that, looks nicer now.